### PR TITLE
split proximity hpp / cpp

### PR DESCRIPTION
**Pull request by j042**
# Overview
splits proximity.hpp to headers and source. Should take less time to compile

### CODE REVIEW

File: src/proximity/proximity.cpp [hunk 1]
```diff
@@ -0,0 +1,288 @@
+#include <napf.hpp>
```
Comment (j042) [inline src/proximity/proximity.cpp:1]: rm

File: splinepy/spline.py [hunk 2]
```diff
@@ -1457,13 +1457,17 @@ def proximities(
 
         queries = _utils.data.enforce_contiguous(queries, dtype="float64")
 
+        # set small tolerance.
+        if tolerance is None and _settings.TOLERANCE > 1.0e-18:
```
Comment (jzwar) [inline splinepy/spline.py:1461]: Interesting

**Review by jzwar — APPROVED** [review]
LGTM

File: splinepy/spline.py (hunk 2 above)
Comment (j042) [reply splinepy/spline.py:1461]: yes

**Pull request merged**